lib/src/settings.rs

```diff
@@ -88,6 +88,18 @@ impl UserSettings {
         "(no email configured)"
     }
 
+    pub fn operation_hostname(&self) -> String {
+        self.config
+            .get_string("operation.hostname")
+            .unwrap_or_else(|_| "(no hostname configured)".to_string())
+    }
+
+    pub fn operation_username(&self) -> String {
+        self.config
+            .get_string("operation.username")
+            .unwrap_or_else(|_| "(no username configured)".to_string())
+    }
+
```

Comment:
Isn't it better to panic instead of falling back to default?

I know user_name/email do that to provide somewhat readable output, but I feel it isn't good idea in general.

Reply:
We definitely shouldn't panic in library code, but returning `Option` here would be better? That way downstream logic can indicate an unset username with unambiguous non-textual means, or panic, or whatever.

Reply:
It's still called in the library, so returning an `Option` wouldn't help. Perhaps we should replace `UserSettings` by a trait that the library user is supposed to implement.

Reply:
Maybe UserSettings constructor can check mandatory parameters and return Result?

Reply:
> It's still called in the library

My thinking is the `unwrap_or_else` can live there instead.

> Perhaps we should replace UserSettings by a trait that the library user is supposed to implement.

This seems excessively complex to me. If we want the caller to pass in settings, why not just use a struct?

> Maybe UserSettings constructor can check mandatory parameters and return Result?

This is good too, if we want to require that these values be specified--though that might not be the best for first time experience.

Reply:
> My thinking is the `unwrap_or_else` can live there instead.

Ah, yes, that makes sense. I'll make that change.

Reply:
Done.